remote/remote_state_s3.go

```diff
@@ -15,6 +15,13 @@ import (
 )
 
 // A representation of the configuration options available for S3 remote state
```

Comment:
Comment needs to be moved to `RemoteStateConfigS3` and the comment here should explain that `ExtendedRemoteStateConfigS3` is used to store custom settings that are not used by Terraform, but by Terragrunt only.

Reply:
Done